### PR TITLE
Add port retries option to kafka-cassandra-streaming script

### PR DESCRIPTION
**Pull request by elezar**
This PR adds the config options from #305 to the kafka_cassandra_streaming script.

As such, it depends on #307 being merged first.

### CODE REVIEW

File: scale-tests/kafka_cassandra_streaming_test.py [hunk 1]
```diff
@@ -40,16 +45,37 @@
...
+
+    --conf-spark-mesos-containerizer <name> [default: mesos]
+    --conf-spark-mesos-driver-failoverTimeout <n> [default: 30]
+    --conf-spark-mesos-executor-home <path> [default: /opt/spark/dist]
+    --conf-spark-scheduler-maxRegisteredResourcesWaitingTime <seconds> [default: 2400s]
+    --conf-spark-scheduler-minRegisteredResourcesRatio <n> [default: 1.0]
+
+    as additional options to make the other values used here configurable.
+    """
+    spark_conf = [
```
Comment (mpereira) [inline scale-tests/kafka_cassandra_streaming_test.py]: Maybe still keep this as a constant being set at the beginning of the script?

File: scale-tests/kafka_cassandra_streaming_test.py [hunk 2]
```diff
@@ -16,18 +16,22 @@
...
-    --producer-number-of-words <n>      number of total words published by producers [default: 1]
+    --producer-number-of-words <n>      number of total words published by producers [default: 0]
     --producer-words-per-second <n>     number of words per second published by producers [default: 1]
     --producer-spark-cores-max <n>      spark.cores.max [default: 2]
     --producer-spark-executor-cores <n> spark.executor.cores [default: 2]
     --consumer-batch-size-seconds <n>   number seconds accumulating entries for each batch request [default: 10]
     --consumer-spark-cores-max <n>      spark.cores.max [default: 1]
     --consumer-spark-executor-cores <n> spark.executor.cores [default: 1]
+
+    --conf-spark-port-maxRetries  <n>          max number of retries for Spark to bind to a port [default: 32]
```
[the remark this answers is not in the document]
Comment (elezar) [reply scale-tests/kafka_cassandra_streaming_test.py]: The tests yesterday were run using `32`: https://github.com/mesosphere/spark-build/blob/master/scale-tests/kafka_cassandra_streaming_test.py#L51

I'm happy to change the default if this makes sense.

[the remark this answers is not in the document]
Comment (susanxhuynh) [reply scale-tests/kafka_cassandra_streaming_test.py]: Hm, it is only a problem if more than 32 drivers land on the same node ... which is unlikely if there are other workloads running in the same cluster. The last time I encountered this limit was by filling an entire cluster with Spark jobs.

**Comment by elezar** [comment]
I'm closing this PR. There are some good ideas, but nothing critical.

**Pull request closed without merging**